contracts/strategies/HStrategy.sol

```diff
@@ -0,0 +1,851 @@
+    /// @param newStrategyParams the new parameters
+    function updateStrategyParams(StrategyParams calldata newStrategyParams) external {
+        _requireAdmin();
+        int24 tickSpacing = pool.tickSpacing();
+        require(
+            newStrategyParams.halfOfShortInterval > 0 &&
+                (newStrategyParams.halfOfShortInterval % tickSpacing == 0) &&
+                newStrategyParams.tickNeighborhood <= newStrategyParams.halfOfShortInterval &&
+                newStrategyParams.tickNeighborhood >= TickMath.MIN_TICK,
+            ExceptionsLibrary.INVARIANT
+        );
+
+        int24 globalIntervalWidth = newStrategyParams.domainUpperTick - newStrategyParams.domainLowerTick;
+        require(
+            (newStrategyParams.domainLowerTick % tickSpacing == 0) &&
+                (newStrategyParams.domainUpperTick % tickSpacing == 0) &&
+                globalIntervalWidth > newStrategyParams.halfOfShortInterval &&
+                (globalIntervalWidth % newStrategyParams.halfOfShortInterval == 0),
+            ExceptionsLibrary.INVARIANT
+        );
+
+        strategyParams = newStrategyParams;
+        emit UpdateStrategyParams(tx.origin, msg.sender, newStrategyParams);
+    }
+
+    /// @notice updates parameters for minting position. Can be called only by admin
+    /// @param newMintingParams the new parameters
+    function updateMintingParams(MintingParams calldata newMintingParams) external {
+        _requireAdmin();
+        require(
+            newMintingParams.minToken0ForOpening > 0 &&
+                newMintingParams.minToken1ForOpening > 0 &&
+                (newMintingParams.minToken0ForOpening <= 1000000000) &&
+                (newMintingParams.minToken1ForOpening <= 1000000000),
+            ExceptionsLibrary.INVARIANT
+        );
+        mintingParams = newMintingParams;
+        emit UpdateMintingParams(tx.origin, msg.sender, newMintingParams);
+    }
+
+    /// @notice updates oracle parameters. Can be called only by admin
+    /// @param newOracleParams the new parameters
+    function updateOracleParams(OracleParams calldata newOracleParams) external {
+        _requireAdmin();
+        require(
+            newOracleParams.averagePriceTimeSpan > 0 &&
+                newOracleParams.maxTickDeviation > 0 &&
+                newOracleParams.maxTickDeviation <= uint24(TickMath.MAX_TICK),
+            ExceptionsLibrary.INVARIANT
+        );
+        oracleParams = newOracleParams;
+        emit UpdateOracleParams(tx.origin, msg.sender, newOracleParams);
+    }
+
+    /// @notice updates parameters of the capital ratios and deviation. Can be called only by admin
+    /// @param newRatioParams the new parameters
+    function updateRatioParams(RatioParams calldata newRatioParams) external {
+        _requireAdmin();
+        require(
+            newRatioParams.erc20CapitalRatioD <= DENOMINATOR &&
+                newRatioParams.minCaptialDeviationD <= DENOMINATOR &&
```

Comment:
might remove this? (this is always true based on two others)
also naming (captial -> capital)